hy/compiler.py

```diff
@@ -2689,18 +2689,20 @@ def hy_compile(tree, module_name, root=ast.Module, get_expr=False):
     body = []
     expr = None
 
-    if tree:
+    if not (isinstance(tree, HyObject) or type(tree) is list):
```

Comment:
If you passed in an object that is neither exactly `list` nor a `HyObject`, the below conditional in the assignment to `spoof_tree` would do the wrong thing.